bigquery_etl/cli/utils.py

```diff
@@ -79,7 +83,25 @@ def table_matches_patterns(pattern, invert, table):
     return matching != invert
 
 
-def paths_matching_name_pattern(pattern, sql_path, project_id, files=["*.sql"]):
+def paths_matching_checks_pattern(pattern, sql_path, project_id):
+    """Return single path to checks.sql matching the name pattern."""
+    checks_files = paths_matching_name_pattern(
+        pattern, sql_path, project_id, ["checks.sql"], CHECKS_FILE_RE
+    )
+    assert len(checks_files) < 2  # should only have 0 or 1 file.
```

Comment:
What if we pattern match on a whole namespace like `firefox_accounts_derived` because we want to run all tests? Is this something we may want to support?

Reply:
We should be careful with using `assert` like this inside application code:
https://stackoverflow.com/a/1838411

Reply:
> What if we pattern match on a whole namespace like firefox_accounts_derived because we want to run all tests? Is this something we may want to support?
This could be added as a Jira item and decided on later, not Phase 1.  Phase 1 is only at the table level for a controlled release.

Reply:
> We should be careful with using assert like this inside application code:

This link is specific to Python and using asserts instead of clean and narrow exception handling.  Using ASSERT here isn't the same use case.  Verification of using ASSERT vs ERROR is one of the many reasons for rolling this out to a small set of non-critical tables.

Reply:
I agree with the first comment, regarding the `assert`. Generally, you have to be careful about using it for logic because if make use of the PYTHONOPTIMIZE assert statements would be entirely removed from the execution code. Meaning if we rely on them to check something and stop the execution then this check could never happen.